src/praisonai/praisonai/agent.py

```
@@ -24,39 +24,23 @@ def __init__(self, *args, cli_backend: Optional[Union[str, Any]] = None, **kwarg
-        """
-        try:
-            from .cli_backends import resolve_cli_backend
-            return resolve_cli_backend(backend_id)
-        except ImportError:
-            raise ImportError(
-                f"CLI backend '{backend_id}' requested but CLI backends not available. "
-                "This may indicate a packaging or installation issue."
-            )
+    def _is_protocol_instance(self, obj) -> bool:
+        """Check if object appears to be a protocol instance (duck typing)."""
+        return hasattr(obj, '__call__') or hasattr(obj, 'process_turn')
```

Ruff (0.15.15)

[warning] Using `hasattr(x, "__call__")` to test if x is callable is unreliable. Use `callable(x)` for consistent results.

Replace with `callable()`

(B004)